django/db/backends/__init__.py

```diff
@@ -823,6 +823,15 @@ def cache_key_culling_sql(self):
         """
         return "SELECT cache_key FROM %s ORDER BY cache_key LIMIT 1 OFFSET %%s"
 
+    def case_cast_sql(self, db_type, internal_type):
```

Comment:
It seems this method will not allow for 3rd party fields to define their own rules. This might be OK for a first cut, but we might later on want to have something better.

I wonder if we could just have a cast_sql with the following API:
```
def cast_sql(self, input_type, output_type):
    """
    Returns a cast expression from input_type to output_type. For example, if input_type is INTEGER
    and output_type is FLOAT, then returns CAST(%s AS FLOAT) for PostgreSQL.

    The return string must have %s placeholder. This will be filled with the input expression,
    which is of type input_type.
    """